.github/workflows/reusable_codespell.yml

```diff
@@ -0,0 +1,38 @@
+name: Codespell
+on:
+  pull_request:
+    branches: [ "main" ]
+  workflow_dispatch:
+    inputs:
+      UMF_BRANCH: 
+        required: true
+      UMF_REPOSITORY:
+        required: true
+        default: https://github.com/oneapi-src/unified-memory-framework.git
+
+permissions:
+  actions: read
+  checks: read
+  contents: read
+  pull-requests: write
+
+jobs:
+  codespell:
+    runs-on: windows-2022
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install codespell
+        run: |
+          pip install codespell
+
+      - name: Run codespell
+        run: |
+          bash .github/scripts/codespell.sh
```

Comment:
if the script is just a few lines you can just put it directly here, no need for extra sh file

Reply:
Script might be useful in case in someone want to use it from their development setup. So instead looking to yml, and copying command, they can just call script which has all correct configuration.

TBH the best solution would be to put this checker in to cmake so we have additional target, similar to cstyle, but script is still beter then yml